src/mrinufft/operators/autodiff.py

```diff
@@ -43,16 +43,21 @@ def backward(ctx, dy):
             ]
             grid_r = torch.meshgrid(*r, indexing="ij")
             grid_r = torch.stack(grid_r, dim=0).type_as(x)[None, ...]
-
             grid_x = x * grid_r  # Element-wise multiplication: x * r
+    
             nufft_dx_dom = torch.cat(
                 [ctx.nufft_op.op(grid_x[:, i, :, :]) for i in range(grid_x.size(1))],
-                dim=1,
+                dim=0,
+            )
+
+            grad_traj = torch.cat(
+                [torch.transpose((-1j * torch.conj(dy[:, i, :]) * nufft_dx_dom[:, i, :]), 0, 1)[None, ...] for i in range(dy.shape[1])],
+                dim=0,
             )
+            
+            grad_traj = torch.mean(grad_traj, dim=0).type_as(traj)
+    
```

Comment:
```suggestion
            grad_traj = torch.mean(
                [torch.transpose((-1j * torch.conj(dy[:, i, :]) * nufft_dx_dom[:, i, :]), 0, 1)[None, ...] for i in range(dy.shape[1])],
                dim=0,
            )    
```

Maybe just have a single line